packages/runtime/container-runtime/src/opLifecycle/opGroupingManager.ts

```diff
@@ -28,6 +28,10 @@ function isGroupContents(opContents: any): opContents is IGroupedBatchMessageCon
 	return opContents?.type === OpGroupingManager.groupedBatchOp;
 }
 
+export function isGroupedBatch(op: ISequencedDocumentMessage): boolean {
+	return isGroupContents(op) || isGroupContents(op.contents);
```

Comment:
This needs a comment, at the very least.  If you can update callsites to be after unpacking, that would be even better IMO.  But I can imagine that may not be trivial.

Reply:
After looking at the usages, it seems like you can know whether it's unpacked yet or not. Take a look, would be better to be explicit/tighter.